Send buffered logs to Loggly after timeout

davidwood opened the pull request:
In cases of low log volume, it's possible to have logs live in the buffer for an extended period of time. I've added an additional parameter that defines a timeout (in ms) after which the buffered logs will automatically be sent to Loggly. The timeout is reset each time a log is created.

I added the timeout as an additional parameter, as opposed to accepting a buffering options object, for backwards compatibility.

davidwood committed:
Added buffer flush timeout

smebberson commented:
This looks good. I'll take a deeper look later this week and merge into the master.

smebberson commented:
Hey @davidwood,

I like what the code does, thanks.

But before I merge the PR, are you happy to tidy up the code a little?
- I use tabs (indented to 4 spaces), rather than spaces
- All if statements should use the long-form of curly braces
- You're not using the force argument of the checkBuffer method, can you please remove it

Thanks.

davidwood committed:
Code style changes

davidwood commented:
Sorry this took so long, I somehow missed the notification. I've switched code indentation over to tabs and updated the if statement in `checkBuffer`. 

The `force` argument in `checkBuffer` is checked on line 58. While this module's call to `checkBuffer` does not use this argument, we've been using it to force a flush of the buffer before a process ends.

MauriceButler commented:
Hi @davidwood 

I have just taken over maintenance of this package, and have just merged in you changes.

I am going to do a bit of cleanup and release a version 1.0.0

Thanks for your pull request.
